conans/client/downloaders/cached_file_downloader.py

```diff
@@ -1,75 +1,68 @@
 import os
 import shutil
+from contextlib import contextmanager
 from threading import Lock
 
-from six.moves.urllib_parse import urlsplit, urlunsplit
-
 from conans.client.downloaders.file_downloader import check_checksum
 from conans.errors import ConanException
 from conans.util.files import mkdir
 from conans.util.locks import SimpleLock
-from conans.util.sha import sha256 as sha256_sum
+from .utils import hash_url
 
 
 class CachedFileDownloader(object):
     _thread_locks = {}  # Needs to be shared among all instances
+    _cache_mapping = '.cached_files'
 
     def __init__(self, cache_folder, file_downloader, user_download=False):
         self._cache_folder = cache_folder
         self._file_downloader = file_downloader
         self._user_download = user_download
 
-    def download(self, url, file_path=None, md5=None, sha1=None, sha256=None, **kwargs):
-        """ compatible interface of FileDownloader + checksum
-        """
-        checksum = sha256 or sha1 or md5
-        # If it is a user download, it must contain a checksum
-        assert (not self._user_download) or (self._user_download and checksum)
-        h = self._get_hash(url, checksum)
-        lock = os.path.join(self._cache_folder, "locks", h)
-        cached_path = os.path.join(self._cache_folder, h)
+    @contextmanager
```

Comment:
Is this something that could be a prior pure-refactor PR?

Reply:
➡️   https://github.com/conan-io/conan/pull/8221